### DISCUSSION TITLE
Support for serverless functions in `build/api` with Github deyployments?

### DISCUSSION THREAD

**Discussion by stevefrench39** [vercel/community#109] [Help] [answered]
TL;DR - We generate serverless functions in a build step and can't get them to work in deployments triggered by Github. Does Vercel support serverless functions contained in a `/build/api` folder or do they have to be in `/api`?

We have a Typescript API with the following file structure:

```
.
├── build [generated during build step]/
│   └── api/
│       └── ping
├── src/
│   └── api/
│       └── ping
└── tsconfig.json
```

On `yarn build`, `src/*` is transpiled to generate the serverless functions in `build/api/*`.  

Deploying this manually is easy.  In Vercel, set `Project Root: build` and run `vercel`.  The serverless functions are successfully built and deployed. However, when Github attempts to automatically deploy a preview branch during a pull request, we receive the following error:

```
Error: The specified Root Directory "build" does not exist. Please update your Project Settings. Learn More: https://vercel.com/docs/build-step#root-directory
```

This is because the `build` directory is not committed to Github, but is generated during the build step.  We tried to fix it by unsetting `Root Directory` and setting `Output Directory: build`.  *In this case, the deployment succeeds it does not build the serverless functions*!

Other than committing our `build` directory to Github, we're out of ideas on how to make this seemingly common use case work.  What are we missing?

**Comment by stevefrench39** [comment, marked as answer]
This was eventually solved by skipping the build step entirely and simply setting `src` as the root directory. Vercel was able to automatically detect and build the serverless functions in the `src/api` directory.
